server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -173,42 +174,43 @@ export default class ServiceProviderReferralsController {
   }
 
   async showInterventionProgress(req: Request, res: Response): Promise<void> {
-    const sentReferral = await this.interventionsService.getSentReferral(
-      res.locals.user.token.accessToken,
-      req.params.id
-    )
+    const { accessToken } = res.locals.user.token
+    const { id } = req.params
+    const sentReferral = await this.interventionsService.getSentReferral(accessToken, id)
+
     const serviceUserPromise = this.communityApiService.getServiceUserByCRN(sentReferral.referral.serviceUser.crn)
     const interventionPromise = this.interventionsService.getIntervention(
-      res.locals.user.token.accessToken,
+      accessToken,
       sentReferral.referral.interventionId
     )
+
     const actionPlanPromise =
       sentReferral.actionPlanId === null
         ? Promise.resolve(null)
-        : this.interventionsService.getActionPlan(res.locals.user.token.accessToken, sentReferral.actionPlanId)
+        : this.interventionsService.getActionPlan(accessToken, sentReferral.actionPlanId)
 
-    const [intervention, actionPlan, serviceUser, supplierAssessment] = await Promise.all([
+    const [intervention, actionPlan, approvedActionPlans, serviceUser, supplierAssessment] = await Promise.all([
       interventionPromise,
       actionPlanPromise,
+      this.interventionsService.getApprovedActionPlanSummaries(accessToken, id),
       serviceUserPromise,
-      this.interventionsService.getSupplierAssessment(res.locals.user.token.accessToken, sentReferral.id),
+      this.interventionsService.getSupplierAssessment(accessToken, sentReferral.id),
     ])
 
+    // appointments should always come from the latest approved action plan
+    const latestApprovedActionPlanSummary = approvedActionPlans.sort(sortBy('-approvedAt'))?.[0] ?? null
```

Comment:
💭 The usage is always `approvedActionPlans.sort(sortBy('-approvedAt'))?.[0]` -- since the older ones are always unused, do we need the list endpoint? Could we have a singular endpoint instead? (YAGNI)

Reply:
this endpoint is used elsewhere, and i think adding a new endpoint just for this use case seems unnecessary.

Reply:
ah did not know -- thanks. Checked its usage and it's used as a list. All good 👍